scene/animation/animation_blend_tree.cpp

```diff
@@ -79,7 +79,11 @@ AnimationNode::NodeTimeInfo AnimationNodeAnimation::get_node_time_info() const {
 
 void AnimationNodeAnimation::_validate_property(PropertyInfo &p_property) const {
 	if (p_property.name == "animation" && get_editable_animation_list) {
-		Vector<String> names = get_editable_animation_list();
+		Vector<String> names;
+		// Prevents off-main-thread errors when generating resource previews.
+		if (is_current_thread_safe_for_nodes()) {
```

Comment:
This is part of engine internals and AFAIK not really meant to be used throughout the codebase. You can see that current usage of this API is limited to core classes: Node and WorkerThreadPool.

So the problem might need a different solution that doesn't make wrong assumptions on thread safety.

CC @RandomShaper @TokageItLab

Reply:
Ah I didn't notice that. It makes sense why you wouldn't want random non-warning thread checks. The only small obvious other thing I can think of is if `get_animation_list` doesn't need to check `!singleton->is_visible()` - it seems like the rest of the method doesn't rely on it as a soft thread check but it does short circuit calls in some events such as reassigning the root node of the tree.

Reply:
`get_editable_animation_list()` is meant to be called only from the main thread because it boils down to a function in some editor plugin (and editor APIs are in general meant to be used from the main thread only). Instead of `is_current_thread_safe_for_nodes()`,, you want to use `Thread::is_main_thread()`. The former worls because the main thread happens to be safe-for-nodes, but both checks aren't conceptually (nor practically) equivalent.

Reply:
Oh ok, I'll update to use that. I originally had that there but wasn't sure if other threads could be marked as safe, and that thread local marker is ultimately what `is_visible` uses (although I'm unsure why I didn't submit this with the exact same call the guard does)